### DISCUSSION TITLE
Can't find mixin - 'Module not found'

### DISCUSSION THREAD

**Discussion by jonmor51** [quasarframework/quasar#12973] [General - Components / Directives / etc] [answered]
Notwithstanding the recommendation to move away from mixins and use the composition API, there's a simple job I want to use a mixin for and I'm familiar with them (from use in non-Quasar Vue projects), so I've tried to use one in a Quasar SPA (with quasar-cli). I've created a 'mixins' directory alongside 'components' etc. in 'src' and put my 'RulesMixin.js' file there. In my component where I want to use it I have:

`import RulesMixin from "mixins/RulesMixin";`

and 

`mixins:[RulesMixin],`

as normal. But the app won't compile, with this error:
```
Module not found: Can't resolve imported dependency "mixins/RulesMixin"
Did you forget to install it? You can run: npm install --save mixins/RulesMixin
```

So is there some special way I have to get mixins to work in Quasar (v2.6)?

**Comment by MilosPaunovic** [comment]
Your path does not seem right, it should be relative, something like `import RulesMixin from "../mixins/RulesMixin"`

**Comment by yusufkandemir** [comment, marked as answer]
The problem here is not about mixins, it's about path alises. There is no alias called `mixins` configured by default, and since you didn't configure it either, it can't be resolved. You basically have 3 options:
1. Use `src` alias: `src/mixins/RulesMixin`
2. Use relative paths: `../mixins/RulesMixin`
3. Add `mixins` alias:
	- For Quasar CLI with Vite: https://quasar.dev/quasar-cli-vite/handling-vite#adding-folder-aliases
	- For Quasar CLI with Webpack: https://quasar.dev/quasar-cli-webpack/handling-webpack#adding-webpack-aliases

**Reply by jonmor51** [reply]
Ah, makes sense now! I figured that as I was treating it the same way as components in terms of paths, it should work the same way. I presume in Vue 2, the webpack alias for 'mixins' was already present, as I don't recall having to configure it myself. I did try `@/mixins/RulesMixin`, without joy, but I hadn't tried `src/mixins/RulesMixin` (which works fine).
